### PR TITLE
[risk=no] [RW-6472] Fitbit, survey, ehr age chart react conversion

### PR DESCRIPTION
**Pull request by SrushtiGangireddy**
Converting age chart of survey, ehr, fitbit view components.

### CODE REVIEW

File: public-ui/src/app/data-browser/charts/chart-biosex/chart-biosex-react.component.tsx [hunk 1]
```diff
@@ -72,9 +72,11 @@ export class BioSexChartReactComponent extends React.Component<Props, State> {
...
                    // Change <= 20 count to display '<= 20'
-                   if (label <= 20) {
-                    return '&#8804; 20';
-                   }
+                   if (series[0].dataOnlyLT20) {
+                    if (label <= 20) {
+                        return '&#8804; 20';
+                    }
+                    }
                    return label;
```
Comment (dolbeew) [inline public-ui/src/app/data-browser/charts/chart-biosex/chart-biosex-react.component.tsx]: Can simplify to a single if statement:
```suggestion
                   if (series[0].dataOnlyLT20 && label <= 20) {
                      return '&#8804; 20';
                    }
                   return label;
```
or even a one-liner:
```suggestion
                   return series[0].dataOnlyLT20 && label <= 20 ? '&#8804; 20' : label;
```

File: public-ui/src/app/data-browser/views/pm/pm.component.ts [hunk 2]
```diff
@@ -127,6 +129,7 @@ export class PhysicalMeasurementsComponent implements OnInit, OnDestroy {
   showMeasurement(group: any, concept: any) {
     this.selectedGroup = group;
     this.selectedConcept = concept;
+    console.log(this.selectedConcept);
```
Comment (dolbeew) [inline public-ui/src/app/data-browser/views/pm/pm.component.ts]: Nit: remove console

File: public-ui/src/app/data-browser/views/pm/pm.component.ts [hunk 3]
```diff
@@ -145,6 +148,7 @@ export class PhysicalMeasurementsComponent implements OnInit, OnDestroy {
         this.setAnalysis();
       }
     }
+    console.log(this.selectedConcept);
```
Comment (dolbeew) [inline public-ui/src/app/data-browser/views/pm/pm.component.ts]: Nit: remove console

File: public-ui/src/app/data-browser/charts/chart-age/chart-age-react.component.tsx [hunk 4]
```diff
@@ -56,6 +63,129 @@ export class AgeChartReactComponent extends React.Component<Props, State> {
...
+  }
+
+  getTooltipHelpText(count, analysisStratumName, percentage, totalCount, domain) {
+      const toolTipHelpText = '<div class="chart-tooltip">' +
+                      '<strong>' + count + '</strong>' + ' participants were ages within range ' +
+                      analysisStratumName + ' when' + (domain === 'pm' ? ' physical measurement with' : '') +
+                      ' this medical concept first occurred and that is <strong>' +
+                      percentage + '</strong>' + '% of all participants with the same criteria. (Total Count = <strong> '
+                      + totalCount + '</strong>) </div>';
+      return toolTipHelpText;
```
Comment (dolbeew) [inline public-ui/src/app/data-browser/charts/chart-age/chart-age-react.component.tsx]: Setting to a const is redundant here, can just return the template:
```suggestion
      return '<div class="chart-tooltip">' +
                      '<strong>' + count + '</strong>' + ' participants were ages within range ' +
                      analysisStratumName + ' when' + (domain === 'pm' ? ' physical measurement with' : '') +
                      ' this medical concept first occurred and that is <strong>' +
                      percentage + '</strong>' + '% of all participants with the same criteria. (Total Count = <strong> '
                      + totalCount + '</strong>) </div>';
```

File: public-ui/src/app/data-browser/charts/chart-age/chart-age-react.component.tsx [hunk 5]
```diff
@@ -56,6 +63,129 @@ export class AgeChartReactComponent extends React.Component<Props, State> {
       this.setState({options: baseOptions});
   }
 
+  getEhrChartOptions() {
+      const {ageAnalysis: {analysisName, results}} = this.props;
```
Comment (dolbeew) [inline public-ui/src/app/data-browser/charts/chart-age/chart-age-react.component.tsx]: Nit: remove unused `analysisName`

File: public-ui/src/app/data-browser/charts/chart-age/chart-age-react.component.tsx [hunk 6]
```diff
@@ -56,6 +63,129 @@ export class AgeChartReactComponent extends React.Component<Props, State> {
...
+      const {categories, series} = this.prepEhrOrSurveyCategoriesAndData(results);
+      this.setCommonAgeChartOptions('Age at First Occurrence in Participant Record',
+      categories, series);
+      this.setState({options: baseOptions});
+  }
+
+  getPMChartOptions() {
+      const {ageAnalysis: {analysisName, results}} = this.props;
```
Comment (dolbeew) [inline public-ui/src/app/data-browser/charts/chart-age/chart-age-react.component.tsx]: Nit: remove unused `analysisName`

File: public-ui/src/app/data-browser/charts/chart-age/chart-age-react.component.tsx [hunk 7]
```diff
@@ -56,6 +63,129 @@ export class AgeChartReactComponent extends React.Component<Props, State> {
...
+      const {categories, series} = this.prepEhrOrSurveyCategoriesAndData(results);
+      this.setCommonAgeChartOptions('Age When Physical Measurement Was Taken', categories, series);
+      this.setState({options: baseOptions});
+  }
+
+  getSurveyChartOptions() {
+      const {ageAnalysis: {analysisName, results}, selectedResult} = this.props;
```
Comment (dolbeew) [inline public-ui/src/app/data-browser/charts/chart-age/chart-age-react.component.tsx]: Nit: remove unused `analysisName`

File: public-ui/src/app/data-browser/charts/chart-age/chart-age-react.component.tsx [hunk 8]
```diff
@@ -56,6 +63,129 @@ export class AgeChartReactComponent extends React.Component<Props, State> {
...
+        seriesName = 'Age When Survey Was Taken';
+    } else {
+        seriesName = 'Age When Physical Measurement Was Taken';
+    }
+    // LOOP CREATES DYNAMIC CHART VARS
+    for (const a of ageAnalysisResults) {
+          // For normal Gender Analysis , the stratum2 is the gender . For ppi it is stratum5;
+          const ageResult = results.filter(x => (domain === 'survey' ? x.stratum2 : x.stratum4) ===
+          (domain === 'survey' ? a.stratum5 : a.stratum2))[0];
+          const count = (a.countValue <= 20) ? '&le; 20' : a.countValue;
```
Comment (dolbeew) [inline public-ui/src/app/data-browser/charts/chart-age/chart-age-react.component.tsx]: Nit: add `toLocaleString()` here also

File: public-ui/src/app/data-browser/charts/chart-age/chart-age-react.component.tsx (hunk 7 above)
Comment (SrushtiGangireddy) [reply public-ui/src/app/data-browser/charts/chart-age/chart-age-react.component.tsx]: Fixed thanks.

File: public-ui/src/app/data-browser/charts/chart-age/chart-age-react.component.tsx (hunk 6 above)
Comment (SrushtiGangireddy) [reply public-ui/src/app/data-browser/charts/chart-age/chart-age-react.component.tsx]: Fixed thanks.

File: public-ui/src/app/data-browser/charts/chart-age/chart-age-react.component.tsx (hunk 5 above)
Comment (SrushtiGangireddy) [reply public-ui/src/app/data-browser/charts/chart-age/chart-age-react.component.tsx]: Fixed thanks.

File: public-ui/src/app/data-browser/charts/chart-age/chart-age-react.component.tsx (hunk 4 above)
Comment (SrushtiGangireddy) [reply public-ui/src/app/data-browser/charts/chart-age/chart-age-react.component.tsx]: Fixed thanks.

File: public-ui/src/app/data-browser/views/pm/pm.component.ts (hunk 3 above)
Comment (SrushtiGangireddy) [reply public-ui/src/app/data-browser/views/pm/pm.component.ts]: Fixed thanks.

File: public-ui/src/app/data-browser/views/pm/pm.component.ts (hunk 2 above)
Comment (SrushtiGangireddy) [reply public-ui/src/app/data-browser/views/pm/pm.component.ts]: Fixed thanks.

File: public-ui/src/app/data-browser/charts/chart-biosex/chart-biosex-react.component.tsx (hunk 1 above)
Comment (SrushtiGangireddy) [reply public-ui/src/app/data-browser/charts/chart-biosex/chart-biosex-react.component.tsx]: Fixed thanks.

**Review by dolbeew — APPROVED** [review]

**Pull request merged**